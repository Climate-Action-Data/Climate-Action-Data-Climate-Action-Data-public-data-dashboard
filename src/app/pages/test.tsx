'use client'
import { type NextPage } from "next";

const Test: NextPage = () => {
    return (
        <div>
            <h1>Hello</h1>
            <p>data</p>
        </div>
    )
}

export default Test