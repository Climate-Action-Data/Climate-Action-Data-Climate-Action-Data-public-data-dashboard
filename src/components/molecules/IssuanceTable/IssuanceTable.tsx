import { DateFormats } from '@/@types/DateFormats'
import { Issuance } from '@/@types/ProjectDetails'
import { ColumnSortFilter } from '@/components/atoms/ColumnSortFilter/ColumnSortFilter'
import { formatDate } from '@/utils/DateFormat'
import { generateRandomString } from '@/utils/GenerationHelpers'
import { Box, Heading, Flex, Table, Tbody, Td, Th, Thead, Tr, Button } from '@chakra-ui/react'
import { useTranslation } from 'react-i18next'

interface IssuanceTableProps {
  issuances: Issuance[]
  onClick: (issuanceId: string) => void
}

export const IssuanceTable = (props: IssuanceTableProps) => {
  const { issuances, onClick } = props
  const { t } = useTranslation(`projectDetails`)

  const handleClick = (issuanceId: string, event?: any) => {
    if (event?.target) {
      const target = event.target as HTMLElement
      if (!(target instanceof HTMLButtonElement)) {
        onClick(issuanceId)
      }
    }
  }

  const renderIssuanceRow = (issuances: Issuance[]) => {
    return issuances.map((issuance) => (
      <Tr data-testid="issuance-row" onClick={(event) => handleClick(issuance.id, event)} key={`issuance-row-${generateRandomString()}`}>
        <Td>{issuance.vintage}</Td>
        <Td>{issuance.quantity.toLocaleString()}</Td>
        <Td>{issuance.availableUnits.toLocaleString()}</Td>
        <Td>{formatDate(issuance.date, DateFormats.YYYY_MM_DD)}</Td>
        <Td>
          <Button variant="hoverOnlyNoBold">{t(`moreDetails`)}</Button>
        </Td>
      </Tr>
    ))
  }

  return (
    <Box minW="840px" maxW="840px" borderRight="1px solid" borderRightColor="lightGray.400">
      <Box borderBottom="2px solid" borderBottomColor="lightGray.700">
        <Flex padding="0 24px" alignItems="center" height="56px">
          <Heading variant={`aeonik`} color="lightGray.700" textTransform="uppercase" as="h3" size="16px">
            {t(`issuances.title`)}
          </Heading>
        </Flex>
      </Box>
      <Box overflowY="scroll" height="329px" maxH="329px">
        <Table variant="simple" className="searchTableSmall">
          <Thead zIndex="4" top="0" position="sticky">
            <Tr>
              <Th>
                <Box>
                  {t(`issuances.vintage`)}
                  <ColumnSortFilter />
                </Box>
              </Th>
              <Th>
                <Box>
                  {t(`issuances.quantity`)}
                  <ColumnSortFilter />
                </Box>
              </Th>
              <Th>
                <Box>
                  {t(`issuances.unitsAvailable`)}
                  <ColumnSortFilter />
                </Box>
              </Th>
              <Th>
                <Box>
                  {t(`issuances.verificationReportDate`)}
                  <ColumnSortFilter />
                </Box>
              </Th>
              <Th>
                <Box />
              </Th>
            </Tr>
          </Thead>
          <Tbody maxH="257px">{renderIssuanceRow(issuances)}</Tbody>
        </Table>
      </Box>
    </Box>
  )
}
