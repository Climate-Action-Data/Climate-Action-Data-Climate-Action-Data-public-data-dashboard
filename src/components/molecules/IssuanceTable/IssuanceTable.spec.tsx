import { fireEvent, render, screen } from '@testing-library/react'
import { IssuanceTable } from './IssuanceTable'
import { PROJECT_DETAIL } from '@/test/TestOvermindMockData'

describe(`RetirementDetails`, () => {
  const mockClick = jest.fn()

  it(`renders correctly issuances`, () => {
    const { container } = render(<IssuanceTable onClick={mockClick} issuances={PROJECT_DETAIL.issuances} />)
    expect(container).toMatchSnapshot()
  })

  it(`renders correctly issuances and row click`, () => {
    const { container } = render(<IssuanceTable onClick={mockClick} issuances={PROJECT_DETAIL.issuances} />)
    const rows = screen.getAllByTestId(`issuance-row`)
    expect(rows.length).toBeGreaterThan(0)
    fireEvent.click(rows[0])
    expect(mockClick).toHaveBeenCalled()
    expect(container).toMatchSnapshot()
  })

  it(`renders correctly with empty array`, () => {
    const { container } = render(<IssuanceTable onClick={mockClick} issuances={[]} />)
    expect(container).toMatchSnapshot()
  })
})
