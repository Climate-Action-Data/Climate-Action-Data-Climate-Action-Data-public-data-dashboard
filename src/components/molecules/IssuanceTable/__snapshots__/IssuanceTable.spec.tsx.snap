// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetirementDetails renders correctly issuances 1`] = `
<div>
  <div
    class="css-ejqwgx"
  >
    <div
      class="css-h9i6sp"
    >
      <div
        class="css-1i18okm"
      >
        <h3
          class="chakra-heading css-zux3ex"
        >
          Issuances
        </h3>
      </div>
    </div>
    <div
      class="css-xsnn5"
    >
      <table
        class="chakra-table searchTableSmall css-0"
      >
        <thead
          class="css-hgvkgs"
        >
          <tr
            class="css-0"
          >
            <th
              class="css-0"
            >
              <div
                class="css-0"
              >
                Vintage
                <div
                  class="chakra-stack css-1yu3riu"
                >
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortAsc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 0.1923L0.191406 4H7.80681L3.99911 0.1923Z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortDesc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 3.8077L0.191406 0H7.80681L3.99911 3.8077Z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </th>
            <th
              class="css-0"
            >
              <div
                class="css-0"
              >
                Quantity
                <div
                  class="chakra-stack css-1yu3riu"
                >
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortAsc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 0.1923L0.191406 4H7.80681L3.99911 0.1923Z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortDesc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 3.8077L0.191406 0H7.80681L3.99911 3.8077Z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </th>
            <th
              class="css-0"
            >
              <div
                class="css-0"
              >
                Units Available
                <div
                  class="chakra-stack css-1yu3riu"
                >
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortAsc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 0.1923L0.191406 4H7.80681L3.99911 0.1923Z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortDesc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 3.8077L0.191406 0H7.80681L3.99911 3.8077Z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </th>
            <th
              class="css-0"
            >
              <div
                class="css-0"
              >
                Verification Report Date
                <div
                  class="chakra-stack css-1yu3riu"
                >
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortAsc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 0.1923L0.191406 4H7.80681L3.99911 0.1923Z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortDesc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 3.8077L0.191406 0H7.80681L3.99911 3.8077Z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </th>
            <th
              class="css-0"
            >
              <div
                class="css-0"
              />
            </th>
          </tr>
        </thead>
        <tbody
          class="css-oznbim"
        >
          <tr
            class="css-0"
            data-testid="issuance-row"
          >
            <td
              class="css-0"
            >
              2010
            </td>
            <td
              class="css-0"
            >
              191
            </td>
            <td
              class="css-0"
            >
              191
            </td>
            <td
              class="css-0"
            >
              2020/03/04
            </td>
            <td
              class="css-0"
            >
              <button
                class="chakra-button css-4xx2wk"
                type="button"
              >
                More Details
              </button>
            </td>
          </tr>
          <tr
            class="css-0"
            data-testid="issuance-row"
          >
            <td
              class="css-0"
            >
              2015
            </td>
            <td
              class="css-0"
            >
              1,102
            </td>
            <td
              class="css-0"
            >
              0
            </td>
            <td
              class="css-0"
            >
              2020/03/04
            </td>
            <td
              class="css-0"
            >
              <button
                class="chakra-button css-4xx2wk"
                type="button"
              >
                More Details
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`RetirementDetails renders correctly issuances and row click 1`] = `
<div>
  <div
    class="css-ejqwgx"
  >
    <div
      class="css-h9i6sp"
    >
      <div
        class="css-1i18okm"
      >
        <h3
          class="chakra-heading css-zux3ex"
        >
          Issuances
        </h3>
      </div>
    </div>
    <div
      class="css-xsnn5"
    >
      <table
        class="chakra-table searchTableSmall css-0"
      >
        <thead
          class="css-hgvkgs"
        >
          <tr
            class="css-0"
          >
            <th
              class="css-0"
            >
              <div
                class="css-0"
              >
                Vintage
                <div
                  class="chakra-stack css-1yu3riu"
                >
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortAsc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 0.1923L0.191406 4H7.80681L3.99911 0.1923Z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortDesc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 3.8077L0.191406 0H7.80681L3.99911 3.8077Z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </th>
            <th
              class="css-0"
            >
              <div
                class="css-0"
              >
                Quantity
                <div
                  class="chakra-stack css-1yu3riu"
                >
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortAsc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 0.1923L0.191406 4H7.80681L3.99911 0.1923Z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortDesc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 3.8077L0.191406 0H7.80681L3.99911 3.8077Z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </th>
            <th
              class="css-0"
            >
              <div
                class="css-0"
              >
                Units Available
                <div
                  class="chakra-stack css-1yu3riu"
                >
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortAsc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 0.1923L0.191406 4H7.80681L3.99911 0.1923Z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortDesc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 3.8077L0.191406 0H7.80681L3.99911 3.8077Z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </th>
            <th
              class="css-0"
            >
              <div
                class="css-0"
              >
                Verification Report Date
                <div
                  class="chakra-stack css-1yu3riu"
                >
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortAsc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 0.1923L0.191406 4H7.80681L3.99911 0.1923Z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortDesc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 3.8077L0.191406 0H7.80681L3.99911 3.8077Z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </th>
            <th
              class="css-0"
            >
              <div
                class="css-0"
              />
            </th>
          </tr>
        </thead>
        <tbody
          class="css-oznbim"
        >
          <tr
            class="css-0"
            data-testid="issuance-row"
          >
            <td
              class="css-0"
            >
              2010
            </td>
            <td
              class="css-0"
            >
              191
            </td>
            <td
              class="css-0"
            >
              191
            </td>
            <td
              class="css-0"
            >
              2020/03/04
            </td>
            <td
              class="css-0"
            >
              <button
                class="chakra-button css-4xx2wk"
                type="button"
              >
                More Details
              </button>
            </td>
          </tr>
          <tr
            class="css-0"
            data-testid="issuance-row"
          >
            <td
              class="css-0"
            >
              2015
            </td>
            <td
              class="css-0"
            >
              1,102
            </td>
            <td
              class="css-0"
            >
              0
            </td>
            <td
              class="css-0"
            >
              2020/03/04
            </td>
            <td
              class="css-0"
            >
              <button
                class="chakra-button css-4xx2wk"
                type="button"
              >
                More Details
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`RetirementDetails renders correctly with empty array 1`] = `
<div>
  <div
    class="css-ejqwgx"
  >
    <div
      class="css-h9i6sp"
    >
      <div
        class="css-1i18okm"
      >
        <h3
          class="chakra-heading css-zux3ex"
        >
          Issuances
        </h3>
      </div>
    </div>
    <div
      class="css-xsnn5"
    >
      <table
        class="chakra-table searchTableSmall css-0"
      >
        <thead
          class="css-hgvkgs"
        >
          <tr
            class="css-0"
          >
            <th
              class="css-0"
            >
              <div
                class="css-0"
              >
                Vintage
                <div
                  class="chakra-stack css-1yu3riu"
                >
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortAsc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 0.1923L0.191406 4H7.80681L3.99911 0.1923Z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortDesc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 3.8077L0.191406 0H7.80681L3.99911 3.8077Z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </th>
            <th
              class="css-0"
            >
              <div
                class="css-0"
              >
                Quantity
                <div
                  class="chakra-stack css-1yu3riu"
                >
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortAsc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 0.1923L0.191406 4H7.80681L3.99911 0.1923Z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortDesc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 3.8077L0.191406 0H7.80681L3.99911 3.8077Z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </th>
            <th
              class="css-0"
            >
              <div
                class="css-0"
              >
                Units Available
                <div
                  class="chakra-stack css-1yu3riu"
                >
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortAsc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 0.1923L0.191406 4H7.80681L3.99911 0.1923Z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortDesc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 3.8077L0.191406 0H7.80681L3.99911 3.8077Z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </th>
            <th
              class="css-0"
            >
              <div
                class="css-0"
              >
                Verification Report Date
                <div
                  class="chakra-stack css-1yu3riu"
                >
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortAsc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 0.1923L0.191406 4H7.80681L3.99911 0.1923Z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    class="chakra-icon css-18gm00e"
                    data-testid="sortDesc"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 3.8077L0.191406 0H7.80681L3.99911 3.8077Z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </th>
            <th
              class="css-0"
            >
              <div
                class="css-0"
              />
            </th>
          </tr>
        </thead>
        <tbody
          class="css-oznbim"
        />
      </table>
    </div>
  </div>
</div>
`;
